app/src/main/java/com/cp/tasky/auth/shared/data/repository/AuthRepositoryImpl.kt

```diff
@@ -61,18 +61,31 @@ class AuthRepositoryImpl(
         }
     }
 
-    override suspend fun registerUser(userRegisterBody: UserRegisterBody) { //TODO: Handle error cases : Result<Nothing, Error>
-        try {
-            remoteDataSource.registerUser(
+    override suspend fun registerUser(userRegisterBody: UserRegisterBody): Result<Unit, Error> {
+        return try {
+            val response = remoteDataSource.registerUser(
                 UserRegisterApiBody(
                     fullName = userRegisterBody.fullName,
                     email = userRegisterBody.email,
                     password = userRegisterBody.password
                 )
             )
+
+            if (response.isSuccessful) {
+                Result.Success(Unit)
+            } else {
+                // We have user inputs validation in-place at FE but existing email can only be validated at BE
+                // We do not want to use message response due to localization support
+                when (response.code()) {
+                    in 400..499 -> Result.Error(error = DataError.Remote.EMAIL_ALREADY_EXIST)
```

Comment:
This is too general, there's a clear status code for when the email exists: `409 Conflict`. 
You'd catch all other client-side errors here with your when expression, like when the API key is invalid, when the request is in a wrong format, when the quota is exceeded, etc.